dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java

```diff
@@ -851,209 +853,219 @@ else if( fieldbcontentname.startsWith("date") ){
+				Boolean isdeleted = con.isArchived();
+				searchResult.put("deleted", isdeleted.toString());
+				Boolean locked = con.isLocked();
+				searchResult.put("locked", locked.toString());
+				searchResult.put("structureInode", con.getStructureInode());
+				searchResult.put("workflowMandatory", String.valueOf(APILocator.getWorkflowAPI().findSchemeForStruct(con.getStructure()).isMandatory()));
+				searchResult.put("contentStructureType", "" + con.getStructure().getStructureType());
 
-			// Workflow Actions
+				// Workflow Actions
 
-			List<WorkflowAction> wfActions = new ArrayList<WorkflowAction>();
+				List<WorkflowAction> wfActions = new ArrayList<WorkflowAction>();
 
-			try {
-				wfActions = APILocator.getWorkflowAPI().findAvailableActions( con, currentUser );
-			} catch ( Exception e ) {
-                Logger.error( this, "Could not load workflow actions : ", e );
-            }
+				try {
+					wfActions = APILocator.getWorkflowAPI().findAvailableActions(con, currentUser);
+				} catch (Exception e) {
+					Logger.error(this, "Could not load workflow actions : ", e);
+				}
 
-			JSONArray wfActionMapList = new JSONArray();
+				JSONArray wfActionMapList = new JSONArray();
 
-			for ( WorkflowAction action : wfActions ) {
-				boolean hasPushPublishActionlet = false;
-                if ( action.requiresCheckout() )
-                    continue;
+				for (WorkflowAction action : wfActions) {
+					boolean hasPushPublishActionlet = false;
+					if (action.requiresCheckout())
+						continue;
 
-                JSONObject wfActionMap = new JSONObject();
-                try {
-					wfActionMap.put( "name", action.getName() );
-	                wfActionMap.put( "id", action.getId() );
-	                wfActionMap.put( "icon", action.getIcon() );
-	                wfActionMap.put( "assignable", action.isAssignable() );
-	                wfActionMap.put( "commentable", action.isCommentable() || UtilMethods.isSet( action.getCondition() ) );
-	                wfActionMap.put( "requiresCheckout", action.requiresCheckout() );
-
-	                List<WorkflowActionClass> actionlets = APILocator.getWorkflowAPI().findActionClasses(action);
-	                for(WorkflowActionClass actionlet : actionlets){
-	                	if(actionlet.getActionlet() != null 
-	                			&& actionlet.getActionlet().getClass().getCanonicalName().equals(PushPublishActionlet.class.getCanonicalName())){
-	                		hasPushPublishActionlet = true;
-	                	}
-	                }
-	                wfActionMap.put( "hasPushPublishActionlet", hasPushPublishActionlet );
-	                try {
-						wfActionMap.put( "wfActionNameStr", LanguageUtil.get( currentUser, action.getName() ) );
-					} catch (LanguageException e) {
-						Logger.error( this, "Could not load language key : " + action.getName() );
-					}
-	                wfActionMapList.add( wfActionMap );
+					JSONObject wfActionMap = new JSONObject();
+					try {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)